pisa/core/detectors.py

```diff
@@ -147,30 +149,33 @@ def get_outputs(self, **kwargs):
         return outputs
 
     def update_params(self, params):
-        for distribution_maker in self:
-            distribution_maker.update_params(params)
-
-        #if None in self.det_names: return # No detector names
-
-        if isinstance(params,Param): params = ParamSet(params) # just for the following
-
-        for p in params.names: # now update params with det_names inside
-            for i, det_name in enumerate(self.det_names):
-                if det_name in p:
-                    cp = deepcopy(params[p])
-                    cp.name = cp.name.replace('_'+det_name, "")
-                    self._distribution_makers[i].update_params(cp)
+        if isinstance(params,Param): params = ParamSet(params)
+            
+        for distribution_maker in self.distribution_makers:
+            ps = deepcopy(params)
```

Comment:
Does `deepcopy` need to be used here? As I understand, it's probably to protect the internal Params whenever what's passed to this method is changed later on, but it messes with `DerivedParams` in a weird little way where they lose references to the original Params they depend on. 

I'm confident I could set it up to update those references when this copy happens, but I thought I'd check to see if there was a specific reason this was implemented as-is!